### PR TITLE
feat(atomic): add parts and options to customize pager icons

### PR DESCRIPTION
**Pull request by mrrajamanickam-coveo**
https://coveord.atlassian.net/browse/KIT-2104

* Add shadow parts and properties to allow customization of Previous and Next button icons in atomic-pager component
* Update Readme with common commands used in dev workflow

## TODO
- [ ] ~~Add/update unit tests~~
- [x] Add/update e2e tests
  - [x] packages/atomic/cypress/e2e/pager.cypress.ts

### CODE REVIEW

File: packages/atomic/src/components/common/pager/pager-common.tsx [hunk 1]
```diff
@@ -42,6 +44,15 @@ export const PagerCommon: FunctionalComponent<PagerProps> = (props) => {
     focusOnFirstResultAndScrollToTop();
   };
 
+  const nextIcon = props.nextIcon || ArrowRight;
+  const prevIcon = props.prevIcon || ArrowRight;
+  const defaultIconStyle = 'w-5 align-middle';
+  // Rotate default next arrow icon by 180 to obtain previous arrow icon
+  const prevIconStyle =
```
Comment (olamothe) [inline packages/atomic/src/components/common/pager/pager-common.tsx:51]: We could also just add an `arrow-left` to our svg assets. It's actually a bit weird that we do not already ship it in our assets ;)

Unless I'm mistaken, the existing arrowRight is a copy of what's available here: https://plasma.coveo.com/foundations/Iconography (arrowHeadRight).

**Review by olamothe — COMMENTED** [review]
Nice touch with the README adjustment !

File: packages/atomic/src/components/common/pager/pager-common.tsx (hunk 1 above)
Comment (mrrajamanickam-coveo) [reply packages/atomic/src/components/common/pager/pager-common.tsx:51]: https://github.com/coveo/ui-kit/blob/KIT-2104/packages/atomic/src/images/arrow-right.svg is slightly different from https://github.com/coveo/plasma/blob/master/packages/style/resources/icons/svg/arrow-right.svg or https://github.com/coveo/plasma/blob/master/packages/style/resources/icons/svg/arrow-right-rounded.svg

Maybe the one in ui-kit is from a different version of plasma ?

Comment (mrrajamanickam-coveo) [reply packages/atomic/src/components/common/pager/pager-common.tsx:51]: Actually [ui-kit's arrow-right-rounded.svg](https://github.com/coveo/ui-kit/blob/KIT-2104/packages/atomic/src/images/arrow-right-rounded.svg) is the same as [plasma's arrow-right-rounded.svg](https://github.com/coveo/plasma/blob/master/packages/style/resources/icons/svg/arrow-right-rounded.svg)

So maybe switch pager to use arrow-right-rounded.svg and add arrow-left-rounded.svg ?

Comment (olamothe) [reply packages/atomic/src/components/common/pager/pager-common.tsx:51]: Yes, we can do that !

Comment (mrrajamanickam-coveo) [reply packages/atomic/src/components/common/pager/pager-common.tsx]: Done

File: README.md [hunk 2]
```diff
@@ -35,18 +37,53 @@ To build a single project for production (for instance, the `@coveo/atomic` pack
...
+
+```sh
+npm run dev:atomic
+```
+
 To start a single project in development (for instance, the `quantic` package), run:
 
 ```sh
-npm dev -w @coveo/quantic
+npm run dev -w @coveo/quantic
```
Comment (btaillon) [inline README.md:51]: We should probably recommend `npx nx` over `npm run -w`.

Either way, Nx will run necessary tasks before the task we asked it to do. However, the last time I tried, Nx's caching mechanism was including the CWD when determining whether a task should be re-run or taken from the cache. Unfortunately, using `npm run -w` seems to change the CWD and invalidate `npm run build`'s cache.

Here's what this task would look like with Nx:
```suggestion
npx nx run quantic:dev
```

Here's what a task with a colon in its name would look like with Nx:
```bash
npx nx run quantic:"test:watch"
```

File: packages/atomic/cypress/e2e/pager.cypress.ts [hunk 3]
```diff
@@ -127,4 +127,45 @@ describe('Pager Test Suites', () => {
...
   });
+
+  describe('Options for previous next button icons', () => {
+    before(() => {
+      new TestFixture().with(addPager()).init();
+    });
+
+    it('should expose shadow part for next icon', () => {
+      cy.get('atomic-pager').shadow().find('[part="next-button-icon"]').click();
+      PagerAssertions.assertRenderPager(2);
```
Comment (btaillon) [inline packages/atomic/cypress/e2e/pager.cypress.ts:138]: This should probably be extracted outside of this `it` block, since it creates another `it` block inside the current `it` block.

File: packages/atomic/cypress/e2e/pager.cypress.ts [hunk 4]
```diff
@@ -127,4 +127,45 @@ describe('Pager Test Suites', () => {
...
+    });
+
+    it('should expose shadow part for previous icon', () => {
+      cy.get('atomic-pager')
+        .shadow()
+        .find('[part="previous-button-icon"]')
+        .click();
+      PagerAssertions.assertRenderPager(1);
```
Comment (btaillon) [inline packages/atomic/cypress/e2e/pager.cypress.ts:146]: This will also create an `it` block in the current `it` block.

File: packages/atomic/cypress/e2e/pager.cypress.ts [hunk 5]
```diff
@@ -127,4 +127,45 @@ describe('Pager Test Suites', () => {
...
+    describe('pager arrow button icon props', () => {
+      const iconTypes = ['previous', 'next'];
+      iconTypes.forEach((iconType) => {
+        it(`should allow ${iconType} icon to be customized`, () => {
+          const iconSelector = `${iconType}-button-icon`;
+          const testCustomIcon =
+            'https://raw.githubusercontent.com/coveo/ui-kit/master/packages/atomic/src/images/arrow-top-rounded.svg';
+
+          new TestFixture()
+            // ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names
```
Comment (btaillon) [inline packages/atomic/cypress/e2e/pager.cypress.ts:158]: No need for this comment, we often do this kind of pattern.

```suggestion
```

**Review by btaillon — APPROVED** [review]
Looks very good!

**Review by y-lakhdar — APPROVED** [review]

**Review by olamothe — APPROVED** [review]

File: packages/atomic/cypress/e2e/pager.cypress.ts [hunk 6]
```diff
@@ -127,4 +127,45 @@ describe('Pager Test Suites', () => {
...
+    before(() => {
+      new TestFixture().with(addPager()).init();
+    });
+
+    it('should expose shadow part for next icon', () => {
+      cy.get('atomic-pager').shadow().find('[part="next-button-icon"]').click();
+      PagerAssertions.assertRenderPager(2);
+    });
+
+    it('should expose shadow part for previous icon', () => {
```
[the remark this answers is not in the document]
Comment (mrrajamanickam-coveo) [reply packages/atomic/cypress/e2e/pager.cypress.ts]: Good suggestion - added an assert.

File: packages/atomic/cypress/e2e/pager.cypress.ts (hunk 4 above)
Comment (mrrajamanickam-coveo) [reply packages/atomic/cypress/e2e/pager.cypress.ts]: Ah, notice now that assertRenderPager has an it - but it doesn't render in the cypress ui. 
Besides assertPagerSelected requires it - errors inside describe.

**Review by mrrajamanickam-coveo — COMMENTED** [review]
@btaillon Thanks for the comments - addressed them, please review.

File: README.md [hunk 7]
```diff
@@ -23,30 +23,72 @@ To install a dependency in a specific package, specify the workspace:
...
+For e2e tests
+```sh
+npx nx run atomic:dev
+# In a separate terminal
+npx nx run atomic:e2e
+```
+
+To run e2e tests for specific files/components using the Cypress GUI
+```sh
+npx nx run atomic:e2e:watch
```
Comment (btaillon) [inline README.md:80]: There is one unfortunate edge-case with Nx: it splits on every `:` character, and assumes that the third item is [an executor configuration](https://nx.dev/recipes/executors/use-executor-configurations).

You can make Nx ignore the second colon symbol by using quotation marks like so:
```suggestion
npx nx run atomic:"e2e:watch"
```

File: packages/atomic/cypress/e2e/pager.cypress.ts [hunk 8]
```diff
@@ -1,3 +1,4 @@
+import {List} from 'lodash';
```
Comment (btaillon) [inline packages/atomic/cypress/e2e/pager.cypress.ts:1]: There's a linter error here which says that this import is never used.

You can apply this suggestion here to remove it:
```suggestion
```

**Review by btaillon — APPROVED** [review]

File: packages/atomic/cypress/e2e/pager.cypress.ts (hunk 8 above)
Comment (mrrajamanickam-coveo) [reply packages/atomic/cypress/e2e/pager.cypress.ts:1]: Thanks for catching this - should have been accidentally created when I was trying to refactor types

File: README.md (hunk 7 above)
Comment (mrrajamanickam-coveo) [reply README.md:80]: works as expected without quotes e.g. it opens cypress in watch mode for following
```
ui-kit % npx nx run atomic:e2e:watch
...
>  NX   Successfully ran target e2e:watch for project atomic
```

**Comment by mrrajamanickam-coveo** [comment]
@jpmarceau  @srosenberg-coveo There are some doc changes in this PR. Can you take a look at them please. Sorry for not including you in the PR before (just learning the process) - If any changes are required I can make a followup PR.

**Comment by jpmarceau** [comment]
> @jpmarceau @srosenberg-coveo There are some doc changes in this PR. Can you take a look at them please. Sorry for not including you in the PR before (just learning the process) - If any changes are required I can make a followup PR.

Looks good, thanks @mrrajamanickam-coveo 👍
